go.mod

```diff
@@ -14,7 +14,7 @@ require (
 	github.com/JeffAshton/win_pdh v0.0.0-20161109143554-76bb4ee9f0ab
 	github.com/MakeNowJust/heredoc v0.0.0-20170808103936-bb23615498cd
 	github.com/Microsoft/go-winio v0.4.11
-	github.com/Microsoft/hcsshim v0.8.6
+	github.com/Microsoft/hcsshim v0.0.0-20190417211021-672e52e9209d
```

Comment:
the hcsshim version `v0.0.0-20190417211021-672e52e9209d` looks strange, the fixed version should be bigger than `v0.8.6`, right?

Reply:
also you need to run following command to update vendor files, only go mod change won't fix this issue.
```
export GO111MODULE=on
go mod vendor
```

Reply:
`hack/update-vendor.sh` will handle the `go mod vendor` and other necessary fixups.

Reply:
Yeah,It's should be bigger than v0.8.6 but the hcsshim repo has not a new tag version

Reply:
> the hcsshim version `v0.0.0-20190417211021-672e52e9209d` looks strange, the fixed version should be bigger than `v0.8.6`, right?

Yeah,It's should be bigger than v0.8.6 but the hcsshim repo has not a new tag version

I don't know why the  pull-kubernetes-dependencies was failed , can you help me ？ any suggestion about that ?

Reply:
> the hcsshim version `v0.0.0-20190417211021-672e52e9209d` looks strange, the fixed version should be bigger than `v0.8.6`, right?

Yeah,It's should be bigger than v0.8.6 but the hcsshim repo has not a new tag version

You mean is that I should run go mod vendor and commit the hcsshim changed files ?

Reply:
@wawa0210 I think we should wait for a new release of `github.com/Microsoft/hcsshim`
about the bazel error, you need to run `hack/update-bazel.sh`

Reply:
No worries.
So if you click [the job failure link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/80116/pull-kubernetes-dependencies/1149994565093036034) from below and expand either the full logs or the "vendor" failure in the junit you will see the diff that caused failure.

At the bottom of that it tells you to run `hack/update-vendor.sh` to fix the failure.

@andyzgangx there's some other non-trivial logic in update-vendor besides go mod vendor and update-bazel (dependency set pruning in particular). It's best to run that tool 😅

Reply:
@andyzhangx  Because the previous version of this repository (v0.8.6) was still five months ago, the code after v0.8.6 was referenced in kubernetes1.14.3.

Reply:
@BenTheElder  i run hack/update-vendor.sh then many other files had been changed ,should i pull all the files have been changed?

Reply:
Should probable also run `hack/pin-dependency.sh $MODULE $SHA-OR-TAG` to pin hcsshim to the new version

Reply:
@feiskyer  Thanks a lot , when i run the script `hack/pin-dependency.sh $MODULE $SHA-OR-TAG` and push again , all the checks passed

Reply:
👍

Reply:
> @andyzhangx Because the previous version of this repository (v0.8.6) was still five months ago, the code after v0.8.6 was referenced in kubernetes1.14.3.

I'm not sure what you're referring to. 1.14 is using 69ac8d3f7fc10a0623f3a2655958a1a5bb71f58f, which is pre-v0.8.4

See https://github.com/kubernetes/kubernetes/blob/release-1.14/Godeps/Godeps.json#L157-L161

```sh
$ pwd
/Users/liggitt/go/src/github.com/Microsoft/hcsshim

$ git tag --contains 69ac8
v0.8.4
v0.8.5
v0.8.6
```

Reply:
> > @andyzhangx Because the previous version of this repository (v0.8.6) was still five months ago, the code after v0.8.6 was referenced in kubernetes1.14.3.
> 
> I'm not sure what you're referring to. 1.14 is using 69ac8d3f7fc10a0623f3a2655958a1a5bb71f58f, which is pre-v0.8.4
> 
> See https://github.com/kubernetes/kubernetes/blob/release-1.14/Godeps/Godeps.json#L157-L161
> 
> ```shell
> $ pwd
> /Users/liggitt/go/src/github.com/Microsoft/hcsshim
> 
> $ git tag --contains 69ac8
> v0.8.4
> v0.8.5
> v0.8.6
> ```

A commit version after v0.8.6 was used before the introduction of go mod. But later, due to the introduction of go mod, the hcsshim version has returned to v0.8.6, which will cause kube-proxy to not work on a specific version of windows server 2019.

Reply:
> A commit version after v0.8.6 was used before the introduction of go mod.

Can you give a link to the kubernetes branch and hcsshim commit you're referring to? The only changes I see to that dependency are at https://github.com/kubernetes/kubernetes/commits/master/vendor/github.com/Microsoft/hcsshim

* a4272099e72ebdaa03da52b98fd28ea861ba0bf9 updated to 69ac8d3f7fc10a0623f3a2655958a1a5bb71f58f, which is pre-v0.8.4
* d0261b10770210f83edbcfe379db24e1b82a9e86 switched to go modules and stayed at the same revision
* c32c26000d68fc0f68d4da920c82ad8701808066 switched to v0.8.6

Reply:
To be clear, I'm not opposed to updating to the current sha if the intent of the hcsshim project is to be used that way... just making sure we're all on the same page that this appears to be a shift in how we're consuming the project, not fixing a regression.

Reply:
> To be clear, I'm not opposed to updating to the current sha if the intent of the hcsshim project is to be used that way... just making sure we're all on the same page that this appears to be a shift in how we're consuming the project, not fixing a regression.

More stringent judgment conditions have been added after v0.8.6, so that this situation no longer occurs

https://github.com/microsoft/hcsshim/commit/672e52e9209d1e53718c1b6a7d68cc9272654ab5

![image](https://user-images.githubusercontent.com/7907809/65444869-8c63f380-de63-11e9-9a84-b2117fd095de.png)

I guess this problem may occur before v0.8.6.